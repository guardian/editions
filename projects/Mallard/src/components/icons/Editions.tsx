import React from 'react';
import Svg, { Circle, Rect } from 'react-native-svg';

const Editions = ({
	height = 42,
	width = 42,
}: {
	height?: number;
	width?: number;
}) => (
	<Svg width={width} height={height} fill="none">
		<Rect x="0.5" y="0.5" width="41" height="41" rx="20.5" stroke="white" />
		<Circle cx="29" cy="14" r="7" fill="#FFBAC8" />
		<Circle cx="14" cy="14" r="7" fill="white" />
		<Circle cx="29" cy="29" r="7" fill="#90DCFF" />
		<Circle cx="14" cy="29" r="7" fill="#FF7F0F" />
	</Svg>
);

export { Editions };
