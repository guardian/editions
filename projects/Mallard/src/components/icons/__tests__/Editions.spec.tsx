import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { Editions } from '../Editions'

describe('Editions', () => {
    it('should display a Editions icon in SVG', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <Editions />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
