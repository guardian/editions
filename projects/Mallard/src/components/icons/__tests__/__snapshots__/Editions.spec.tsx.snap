// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editions should display a Editions icon in SVG 1`] = `
<RNSVGSvgView
  bbHeight="42"
  bbWidth="42"
  fill="none"
  focusable={false}
  height={42}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      Object {
        "flex": 0,
        "height": 42,
        "width": 42,
      },
    ]
  }
  width={42}
>
  <RNSVGGroup
    fill={null}
    propList={
      Array [
        "fill",
      ]
    }
  >
    <RNSVGRect
      fill={
        Object {
          "payload": 4278190080,
          "type": 0,
        }
      }
      height="41"
      propList={
        Array [
          "stroke",
        ]
      }
      rx="20.5"
      stroke={
        Object {
          "payload": 4294967295,
          "type": 0,
        }
      }
      width="41"
      x="0.5"
      y="0.5"
    />
    <RNSVGCircle
      cx="29"
      cy="14"
      fill={
        Object {
          "payload": 4294949576,
          "type": 0,
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      r="7"
    />
    <RNSVGCircle
      cx="14"
      cy="14"
      fill={
        Object {
          "payload": 4294967295,
          "type": 0,
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      r="7"
    />
    <RNSVGCircle
      cx="29"
      cy="29"
      fill={
        Object {
          "payload": 4287683839,
          "type": 0,
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      r="7"
    />
    <RNSVGCircle
      cx="14"
      cy="29"
      fill={
        Object {
          "payload": 4294934287,
          "type": 0,
        }
      }
      propList={
        Array [
          "fill",
        ]
      }
      r="7"
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
